tools/tdbg/commands.go

```diff
@@ -180,9 +180,51 @@ func describeMutableState(c *cli.Context) (*adminservice.DescribeMutableStateRes
 	return resp, nil
 }
 
-// AdminDeleteWorkflow delete a workflow execution from Cassandra and visibility document from Elasticsearch.
+// AdminDeleteWorkflow force delete a workflow's mutable state, history, and visibility records as long as it's possible.
+// It should only be used as a troubleshooting tool since no additional check will be done before the deletion.
+// (e.g. if a child workflow has recorded its result in the parent workflow)
+// Please use normal workflow delete command to gracefully delete a workflow execution.
 func AdminDeleteWorkflow(c *cli.Context) error {
-	return fmt.Errorf("not implemented")
+	adminClient := cFactory.AdminClient(c)
+
+	namespace, err := getRequiredOption(c, FlagNamespace)
+	if err != nil {
+		return err
+	}
+	wid, err := getRequiredOption(c, FlagWorkflowID)
+	if err != nil {
+		return err
+	}
+	rid := c.String(FlagRunID)
```

Comment:
Should RunID also be required in this case?

Reply:
hmm I think we should support operate only based on workflowID to be consistent with all other commands. Could be useful if only workflowID is known or operator knows the target workflow is open. 

There's a confirmation step so I am not quite worry about operator forgets to specify the runID and accidentally deleted the current run.